src/database/models/student-grade-instance.ts

```diff
@@ -0,0 +1,210 @@
+import { Model, DataTypes, BelongsToGetAssociationMixin, HasManyGetAssociationsMixin } from 'sequelize';
+import appSequelize from '../app-sequelize';
+import * as _ from 'lodash';
+
+export interface StudentGradeInstanceQuestionOverridesInterface {
+    webworkQuestionPath: string;
+    problemNumber: number;
+}
+
+export interface StudentGradeInstanceGradeOverridesInterface {
+    randomSeed: number;
+    // numAttempts: number;
+    // This is a jsonb field so it could be any (from db)
+    // Submitted in workbook used any so I'm going to keep it consistent here
+    // If this is used for form data we will never know any info about what keys are available
+    // Might make sense to make this an unknown type since I don't think we will ever access the types
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    currentProblemState: any;
+}
+
+export interface StudentGradeInstanceInterface extends StudentGradeInstanceQuestionOverridesInterface, StudentGradeInstanceGradeOverridesInterface {
+    id: number;
+    studentGradeId: number;
+    // userId: number;
+    // courseWWTopicQuestionId: number;
+    studentTopicAssessmentInfoId: number;
+    scoreForBestVersion: number; // the score from the highest-scoring exam submission
+    overallBestScore: number; // the best score on this problem alone
+    // Grade instances shouldn't be overwritten
+    // effectiveScore: number;
+    // Doesn't have dead date
+    // legalScore: number;
+    // Don't need other num attempts fields because there is no doing so after the fact?
+    // numAttempts: number;
+    // Don't need locked because it is very controlled
+    // Don't need references to workbooks since there should be a relatively 1-1 relationship
+    active: boolean;
+    bestIndividualAttemptId: number;
+    bestVersionAttemptId: number;
+}
+
+export default class StudentGradeInstance extends Model implements StudentGradeInstanceInterface {
+    public id!: number;
+    public studentGradeId!: number;
+    // public userId!: number;
```

Comment:
why is this commented out? Do we need it for the association?

Reply:
Good catch
This part is just decorative for typescript, it gives no functionality.
It would only be a problem if we wanted to access the userId

That being said though your right, why are we commenting this out

Reply:
grade instance -> student grade -> user or grade instance -> student topic assessment info -> user

there are several fields commented out here, not just userId, we should dump them all